pkg/util/azureclient/storage/storage.go

```diff
@@ -90,6 +112,27 @@ type Blob interface {
 
 type blob struct {
 	*storage.Blob
+	log *logrus.Entry
 }
 
 var _ Blob = &blob{}
+
+func (b *blob) CreateBlockBlobFromReader(blob io.Reader, options *storage.PutBlobOptions) error {
+	ctx := context.Background()
+	data, err := ioutil.ReadAll(blob)
+	if err != nil {
+		return err
+	}
+	return wait.PollImmediateUntil(3*time.Second,
```

Comment:
this is not retrying 3 times, it is retrying indefinitely every 3 seconds.  Please follow the pattern in, e.g. `func (rs *retrySender) Do`

Reply:
ok, I agree about the retrying 3 times, but the retrySender.Do() seems to be at the wrong level (too lower level) as it is going to make resetting the Reader() impossible.